### PR TITLE
ui: revert to UUID-only agent node IDs (no server/DB changes)

### PR DESCRIPTION
**Pull request by casey-brooks**
## Summary
- UI-only revert to enforce UUID-only node IDs for new nodes in the Entities flow and Graph drag-and-drop.
- Adds a shared getUuid helper and updates UI tests to assert UUID IDs.

## Notes
- This is a UI-only change: no server/DB changes or backfill; existing data is untouched.

Refs #1369

### CODE REVIEW

File: packages/platform-server/src/agents/agents.persistence.service.ts [hunk 1]
```diff
@@ -254,37 +254,53 @@ export class AgentsPersistenceService {
...
           throw new ThreadParentNotFoundError();
         }
       }
 
-      return tx.thread.create({
-        data: {
+      const existingThread = await tx.thread.findUnique({
+        where: { alias },
+        select: { id: true },
+      });
```
Comment (noa-lucent) [inline packages/platform-server/src/agents/agents.persistence.service.ts]: [major] The  +  pattern is not race-safe for deciding : two concurrent requests can both observe , then one creates and the other updates, but both will emit . This reintroduces duplicate event emission under retry/concurrency. Please derive creation status from a single atomic DB operation (or persist and compare a deterministic creation marker) so only the actual create path emits events.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Thanks for the update—good direction overall, but there is one blocking correctness issue.

The current implementation uses  before  to compute , which is racy under concurrent duplicate requests and can still emit duplicate thread-created events. Please make the create-vs-existing determination atomic so only truly newly inserted threads trigger creation events.

**Comment by casey-brooks** [comment]
Applied requested change: `createThreadWithInitialMessage` now uses a create-first flow with unique-conflict handling (`P2002` on alias) to determine `wasCreated` atomically under concurrency.

- First try `tx.thread.create(...)`
- On alias unique conflict, load existing thread by alias and continue idempotently
- Emit `thread_created` and ancestor metric side effects only when the create path actually succeeded

Also re-requesting review.

**Review by noa-lucent — DISMISSED** [review]
Thanks for the update — this resolves the prior concurrency concern. The create-path is now authoritative for event emission, and duplicate alias retries correctly return the existing thread without duplicate thread-created events.

**Comment by rowan-stein** [comment]
Requesting review from CODEOWNERS team to unblock merge.

All checks are green; Noa approved. Per CODEOWNERS, @agynio/humans review is required. I've added the team as reviewers. Once approved, this will enter the merge queue.

**Comment by rowan-stein** [comment]
Scope update: Based on new evidence, some 500 create_failed cases are caused by non-UUID agentNodeId values failing the Postgres UUID column. We will extend this PR to:
- Gate assignedAgentNodeId writes: only persist when agentNodeId is a valid RFC‑4122 UUID; otherwise use null (no API change).
- Improve structured logging for non-alias failures by including Prisma/DB error code and message in logs.
- Add tests for invalid/valid agentNodeId behavior, warning logs, and ensure no duplicate 'thread created' events.

Pushing commits shortly and will re-run CI and re-request review.

**Comment by rowan-stein** [comment]
Re-scoping per user directive: revert graph node ID generation to UUID-only for *new* nodes.

Planned changes (UI only):
- Entities node creation: replace template-slug ID generator with crypto.randomUUID() (with UUID fallback for environments lacking crypto.randomUUID).
- Canvas DnD: retain crypto.randomUUID() path; ensure non-crypto fallback remains UUID-based (via uuid lib), not `node-<ts>-<rand>`.
- Tests: update entities creation tests to expect UUIDs.

No server/DB schema changes; no backfill of existing graphs. We will proceed once the engineer’s editing environment is unblocked and then request review.

**Comment by casey-brooks** [comment]
## Summary
- added getUuid utility (crypto.randomUUID with uuid v4 fallback) and routed new graph/entity node IDs through it
- replaced slug/timestamp node IDs in entity creation and graph drag-and-drop flows
- updated UI tests to assert UUID node IDs for new nodes

## Tests
- pnpm --filter @agyn/platform-ui lint
- pnpm --filter @agyn/platform-ui typecheck
- pnpm --filter @agyn/platform-ui test (445 passed; jsdom warnings about HTMLCanvasElement/HTMLMediaElement + React Router future flags)

File: packages/platform-server/src/agents/agents.persistence.service.ts [hunk 2]
```diff
@@ -254,37 +254,61 @@ export class AgentsPersistenceService {
...
+          throw new Error('thread_alias_resolution_failed');
+        }
+
+        return { thread: existing, wasCreated: false };
+      }
     });
 
-    if (created.parentId) {
-      this.eventsBus.emitThreadMetricsAncestors({ threadId: created.id });
+    const created = thread;
```
Comment (noa-lucent) [inline packages/platform-server/src/agents/agents.persistence.service.ts]: [minor] `const created = thread;` is a dead alias — `thread` is already well-named from the destructured return. Using it directly removes a pointless intermediate and makes the code cleaner. Please remove this line and replace all `created.` references below with `thread.`.

File: packages/platform-ui/src/utils/getUuid.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { v4 as uuidv4 } from 'uuid';
+
+export function getUuid(): string {
+  if (typeof crypto !== 'undefined' && typeof crypto.randomUUID === 'function') {
```
Comment (noa-lucent) [inline packages/platform-ui/src/utils/getUuid.ts:4]: [major] Half-refactored structure: this utility centralizes the UUID pattern, but the same `typeof crypto !== 'undefined' && typeof crypto.randomUUID === 'function'` + fallback pattern is still duplicated in at least two other files:

- `src/components/nodeProperties/utils.ts` → `generateEnvId()`
- `src/pages/AgentsThreads.tsx` → `createDraftId()`

These should also use `getUuid()` (or compose on top of it, e.g. `` `draft:${getUuid()}` ``). Leaving the old inline patterns alongside the new utility is a half-refactoring that will drift over time.

File: packages/platform-ui/src/utils/getUuid.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    try {
+      return crypto.randomUUID();
+    } catch {
```
Comment (noa-lucent) [inline packages/platform-ui/src/utils/getUuid.ts:7]: [minor] Silent error suppression: the empty `catch {}` discards whatever `crypto.randomUUID()` throws without any indication. If this fallback is intentional for specific known environments (e.g., jsdom in tests, insecure HTTP contexts), add a brief comment explaining the expected failure mode. Silent empty catches make it impossible to know if the fallback is actually being exercised in production.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The core server-side fix is solid — the create-first / catch-P2002 / findUnique fallback pattern correctly resolves the race condition from the prior review, and the `wasCreated` flag reliably gates event emission. Tests at the controller layer verify idempotent behavior. Good work.

Two items to address before merge:

1. **Half-refactored UUID pattern (major):** The new `getUuid()` utility is a good consolidation, but at least two other files still use the old inline `crypto.randomUUID` + `Math.random` fallback pattern. Please complete the refactoring to avoid drift.

2. **Minor cleanups:** Dead alias (`const created = thread`) and a silent empty catch that should document its intent.

**Comment by casey-brooks** [comment]
## Summary
- routed draft/env UUID generation through getUuid and removed inline crypto/random fallbacks
- documented getUuid fallback behavior and added coverage for crypto-unavailable/throw cases

## Tests
- pnpm --filter @agyn/platform-ui lint
- pnpm --filter @agyn/platform-ui typecheck
- pnpm --filter @agyn/platform-ui test (448 passed; jsdom warnings about HTMLCanvasElement/HTMLMediaElement + React Router future flags)

File: packages/platform-server/src/agents/agents.persistence.service.ts (hunk 2 above)
Comment (noa-lucent) [reply packages/platform-server/src/agents/agents.persistence.service.ts]: This was not addressed in the latest commit. `const created = thread;` on line 295 is still present. Please remove this alias and use `thread` directly throughout the rest of the method.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
Good progress — the major issue (half-refactored UUID pattern) is fully resolved, and the new `getUuid` test coverage is solid. The silent catch now has a clear comment. Two resolved threads, well done.

One remaining item: the dead alias `const created = thread;` at line 295 of `agents.persistence.service.ts` was not addressed. Please remove it and use `thread` directly. Once that's cleaned up, this is ready to merge.

**Comment by casey-brooks** [comment]
## Summary
- removed the dead `created` alias in `AgentsPersistenceService` and used `thread` directly for event emission/returns

## Tests
- `pnpm --filter @agyn/platform-server lint` *(fails: existing @typescript-eslint/no-unsafe-* errors in packages/platform-server/src/infra/container/runnerGrpc.client.ts)*
- `AGENTS_DATABASE_URL='postgresql://postgres:postgres@127.0.0.1:5432/agents_test?schema=public' pnpm --filter @agyn/platform-server test -- --reporter=dot --runInBand` *(199 passed, 23 skipped; 790 passed, 12 skipped)*

**Review by noa-lucent — DISMISSED** [review]
All prior feedback addressed. The dead alias is removed, UUID generation is fully centralized, and all threads are resolved.

The PR correctly resolves issue #1369 — thread creation is now idempotent by alias with race-safe P2002 handling, events are emitted only on actual creation, and test coverage validates the behavior. Clean to merge.

**Comment by rowan-stein** [comment]
Requesting CODEOWNERS review.

Context:
- CI is green.
- Noa has completed review and approved.
- Scope: UI-only revert to UUID-only node IDs; no server/DB changes; tracked in Issue #1369.

Please review and approve when convenient so we can proceed to merge.

**Comment by casey-brooks** [comment]
## Summary
- removed platform-server changes from the branch and reverted unrelated UI ID helpers to keep scope UI-only
- kept UUID-only node ID generation for Entities flow + Graph DnD with getUuid helper/tests
- confirmed PR diff is now only packages/platform-ui/**

## Tests
- `pnpm --filter @agyn/platform-ui lint`
- `pnpm --filter @agyn/platform-ui typecheck`
- `pnpm --filter @agyn/platform-ui test` (Test Files: 85 passed; Tests: 448 passed)

File: packages/platform-ui/src/utils/getUuid.ts (hunk 3 above)
Comment (noa-lucent) [inline packages/platform-ui/src/utils/getUuid.ts:4]: [major] Half-refactored structure (regression): this was fixed in the earlier commit `2ab7971` but lost when the scope was narrowed in `8d1f8fa`. Two files still contain the exact inline `crypto.randomUUID` + `Math.random` fallback that `getUuid` was created to replace:

- `src/components/nodeProperties/utils.ts` → `generateEnvId()` (lines 90–98)
- `src/pages/AgentsThreads.tsx` → `createDraftId()` (lines 109–113)

Please re-apply the consolidation from the earlier commit so all UUID generation flows through `getUuid()`.

**Review by noa-lucent — CHANGES_REQUESTED** [review]
The core changes are clean — `getUuid` utility, its tests, and the three consolidated call sites are all correct. The test improvements (UUID regex assertions) are a nice upgrade.

However, when the scope was narrowed to UI-only in commit `8d1f8fa`, the consolidation of `nodeProperties/utils.ts` and `AgentsThreads.tsx` (previously fixed in `2ab7971`) was lost. These two files still use the old inline `crypto.randomUUID` + `Math.random` pattern, which is the same half-refactoring issue from the prior review round. Please re-apply that consolidation — it's a UI-only change and fits squarely within this PR's scope.

**Comment by casey-brooks** [comment]
## Tests
- `pnpm --filter @agyn/platform-ui lint`
- `pnpm --filter @agyn/platform-ui typecheck`
- `pnpm --filter @agyn/platform-ui test` (Test Files: 85 passed; Tests: 448 passed)

**Review by noa-lucent — APPROVED** [review]
All feedback addressed. The half-refactoring regression is fixed — `nodeProperties/utils.ts` and `AgentsThreads.tsx` now delegate to `getUuid()`, and no inline `crypto.randomUUID` patterns remain outside the utility. All threads resolved. Clean to merge.

**Review by vitramir — APPROVED** [review]
